packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+      return true;
+    }());
+    return result;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    menuScopeNode = FocusScopeNode();
+    assert(() {
+      menuScopeNode.debugLabel = '$MenuButton(Scope for ${widget.label})';
+      return true;
+    }());
+    _buttonFocusNode.addListener(_handleFocusChange);
+  }
+
+  @override
+  void dispose() {
+    menuScopeNode.dispose();
+    entry.dispose();
+    _internalFocusNode?.removeListener(_handleFocusChange);
+    _internalFocusNode?.dispose();
+    _internalMenuController?.dispose();
+    _internalFocusNode = null;
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _updateChildMenu();
+  }
+
+  @override
+  void didUpdateWidget(MenuButton oldWidget) {
+    if (widget.focusNode != oldWidget.focusNode) {
+      if (oldWidget.focusNode == null) {
+        _internalFocusNode?.removeListener(_handleFocusChange);
+        _internalFocusNode?.dispose();
+        _internalFocusNode = null;
+      } else {
+        oldWidget.focusNode!.removeListener(_handleFocusChange);
+      }
+      _buttonFocusNode.addListener(_handleFocusChange);
+    }
+    _updateChildMenu();
+    super.didUpdateWidget(oldWidget);
+  }
+
+  void _updateChildMenu() {
+    final MenuController controller = MenuController.maybeOf(context) ?? (_internalMenuController ??= MenuController());
+    final _MenuNode parent = _MenuNode.maybeOf(context) ?? controller._root;
+    final AlignmentGeometry menuAlignment;
+    switch (parent.orientation) {
+      case Axis.horizontal:
+        menuAlignment = widget.alignment ?? AlignmentDirectional.bottomStart;
+        break;
+      case Axis.vertical:
+        menuAlignment = widget.alignment ?? AlignmentDirectional.topEnd;
+        break;
+    }
+
+    final MenuThemeData menuTheme = MenuTheme.of(context).copyWith(
+      menuBackgroundColor: widget.backgroundColor,
+      menuShape: widget.shape,
+      menuPadding: widget.padding,
+    );
+
+    final EdgeInsetsDirectional menuPadding =
+        menuTheme.menuPadding ?? MenuTheme.of(context).menuPadding ?? _TokenDefaultsM3(context).menuPadding;
+
+    final Offset menuPaddingOffset;
+    switch (parent.orientation) {
+      case Axis.horizontal:
+        menuPaddingOffset = widget.alignmentOffset ?? Offset(-menuPadding.start, 0);
+        break;
+      case Axis.vertical:
+        menuPaddingOffset = widget.alignmentOffset ?? Offset(0, -menuPadding.top);
+        break;
+    }
+
+    if (childMenu == null) {
+      entry = _ChildMenuNode(
+        buttonFocusNode: _buttonFocusNode,
+        parent: parent,
+        controller: controller,
+        menuScopeNode: menuScopeNode,
+        onOpen: widget.onOpen,
+        onClose: widget.onClose,
+        alignment: menuAlignment,
+        alignmentOffset: menuPaddingOffset,
+        widgetChildren: widget.children,
+      );
+      childMenu = _createMenuEntryFromExistingNode(entry);
+    }
+    entry
```

Comment:
if the menu was just created in the above if, this configuration work is happening twice... maybe use an else?